spec/helpers/mock_project.rb

```diff
@@ -31,10 +31,16 @@ def require_mock_gem(gem, require: nil)
     sig { params(name: String, version: T.nilable(String), require: T.nilable(T.any(FalseClass, String))).void }
     def require_real_gem(name, version = nil, require: nil)
       line = +"gem \"#{name}\""
+
+      gemfile = read("Gemfile")
+      lines = gemfile.lines.delete_if { |gemfile_line| gemfile_line.start_with?(line) }
```

Comment:
This is not ideal. We'll spend a loot of time reading the Gemfile for each test suite.
It seems this is only required for the `check-shim` test where we pass a custom version of the Sorbet right?

What do you think about removing this and creating the project manually in the `check-shims` test instead?

```rb
@project = MockProject.new("#{TEST_TMP_PATH}/#{spec_name}/project")
@project.gemfile(project.tapioca_gemfile)
```

We could even extract `#{TEST_TMP_PATH}/#{spec_name}/project` as a service to make it easier.

Reply:
Ok, I changed this to make the `mock_project` method take an optional `sorbet_dependency` keyword parameter that is `true` by default, and making just this test reinitialize the mock project with that parameter value passed as `false`.